.travis.yml

```diff
@@ -0,0 +1,70 @@
+language: cpp
+
+matrix:
+  include:
+    # Linux, default compilers
+    - os: linux
+    # Linux, latest supported compilers
+    - os: linux
+      env: GCC=gcc-6 CLANG=clang-6.0
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-6.0
+          packages:
+            - gcc-6
+            - g++-6
+            - clang-6.0
+    # Linux, AddressSanitizer
+    - os: linux
+      env: CFG=asan
+      sudo: true # LeakSanitizer doesn't work inside Travis containers, see https://github.com/travis-ci/travis-ci/issues/9033
+    # Linux, MemorySanitizer
+    - os: linux
+      env: CFG=msan
+    # OSX, default compilers, disabled because some tests in CoreTest are consistently failing and FBuildTest times out.
+    #- os: osx
+    #  compiler: clang
+  allow_failures:
+    - env: CFG=asan
+    - env: CFG=msan
+    - os: osx
+  fast_finish: true
+
+script:
+  - cd Code
+  # Set up some variables for convenience
+  - export NPROC=$(if [[ "${TRAVIS_OS_NAME}" == osx ]]; then sysctl -n hw.ncpu; else nproc; fi)
+  - export MAKE_CMD="make -j${NPROC}"
+  - export FBUILD_CMD="./fbuild -j${NPROC} -noprogress"
+  - export GCC=${GCC:-gcc} CLANG=${CLANG:-clang}
+  - export GXX=${GCC/gcc/g++} CLANGXX=${CLANG/clang/clang++}
+  # Patch configs to use installed compilers
+  - |
+    sed -ie "1i\\
+    #define CI_BUILD
```

Comment:
This CI_BUILD define doesn't seem to be visible to FASTBuild. I also don't understand why the bff changes would be needed.

Reply:
This `sed` command inserts this define at the top of `fbuild.bff` and later when this define is present hard-coded paths to compiler binaries are replaced with paths that exist in the current environment.  
But since we now have dedicated files for compiler configuration this logic can be done much cleaner.

As to why this is needed at all - it is because in each Linux distribution compilers have different versioned names and stored in different places (especially `cc1` and `cc1plus` binaries).